src/types/UiaTextRangeBase.cpp

```diff
@@ -1013,7 +1013,9 @@ std::wstring UiaTextRangeBase::_getTextValue(til::CoordType maxLength) const
         }
     }
 
-    if (maxLength >= 0)
+    // if the caller asked for INT_MAX,
+    // they probably just want the whole thing
+    if (maxLength >= 0 && maxLength != INT_MAX)

```

Comment:
@lhecker is there some integer promotion rule that would result in -1 being converted to `INT_MAX`

Reply:
Carlos explained it to me. I think we should make sure we do length checks inside the loop above on 1010.